src/NServiceBus.Core/DelayedDelivery/TimeoutManager/TimeoutManager.cs

```diff
@@ -56,22 +56,22 @@ static void SetupTimeoutPoller(FeatureConfigurationContext context, string dispa
                     waitTime,
                     ex => criticalError.Raise("Repeated failures when fetching timeouts from storage, endpoint will be terminated.", ex));
 
-                return new ExpiredTimeoutsPoller(b.Build<IQueryTimeouts>(), b.Build<IDispatchMessages>(), dispatcherAddress, circuitBreaker, () => DateTime.UtcNow);
+                return new ExpiredTimeoutsPoller(b.Build<IQueryTimeouts>(), context.Transport.Dispatcher, dispatcherAddress, circuitBreaker, () => DateTime.UtcNow);
```

Comment:
Not sure it is needed to expose the dispatcher like this. If we'd expose the TransportInfrastructure directly you could also call the factory to create a dispatcher and then the transport can decide what lifetime the dispatcher has which I think has benefits.